include/itkSpectra1DNormalizeImageFilter.hxx

```diff
@@ -0,0 +1,126 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkSpectra1DNormalizeImageFilter_hxx
+#define itkSpectra1DNormalizeImageFilter_hxx
+
+
+#include "itkImageScanlineIterator.h"
+#include "itkTotalProgressReporter.h"
+
+namespace itk
+{
+template <typename TInputImage, typename TReferenceImage>
+void
+Spectra1DNormalizeImageFilter<TInputImage, TReferenceImage>::GenerateInputRequestedRegion()
+{
+  // call the superclass' implementation of this method
+  Superclass::GenerateInputRequestedRegion();
+
+  // ImageToImageFilter expects all inputs to be of TInputImage type
+  // so we don't use this->GetInput(), instead we use ProcessObject directly
+  auto * in1 = reinterpret_cast<const ReferenceImageType *>(this->ProcessObject::GetInput(1));
+  auto * reference = const_cast<ReferenceImageType *>(in1);
+  // instead of fiddling with cropping the requested region to reference line,
+  // we simply request the entire reference line
+  reference->SetRequestedRegionToLargestPossibleRegion();
+}
+
+#define DeclareVectorDivideOperator(TypeA, TypeB)                      \
```

Comment:
This is element-wise divide. Other ways to divide are possible. Absence of divide operator from both Vector and VLV is probably a design choice. I think this should stay here.

Reply:
If ITK is built with ITKUltrasound then this divide operator will be added to the itk namespace. If it is a design choice to avoid implementing a divide operator in ITK, would it be more reasonable to define a free function here that would be harder to misuse by accident? Something like `divide()`?

Reply:
Good point. Will do.